server/queue/queue.php

```diff
@@ -39,24 +39,32 @@ function getAppointments($date, $siteId) {
 	try {
 		$canViewClientInformation = $USER->hasPermission('view_client_information');
 
-		$query = 'SELECT Appointment.appointmentId, TIME_FORMAT(AppointmentTime.scheduledTime, "%l:%i %p") AS scheduledTime, 
+		$query = 'SELECT t1.appointmentId, TIME_FORMAT(AppointmentTime.scheduledTime, "%l:%i %p") AS scheduledTime, 
 					firstName, lastName, timeIn, timeReturnedPapers, timeAppointmentStarted, timeAppointmentEnded, 
 					completed, cancelled, language, Client.clientId, 
 					(DATE_ADD(AppointmentTime.scheduledTime, INTERVAL 30 MINUTE) < NOW() AND timeIn IS NULL) AS noShow,
-					(AppointmentTime.scheduledTime < Appointment.createdAt) AS walkIn ';
+					(AppointmentTime.scheduledTime < t1.createdAt) AS walkIn , appointmentType.name as appointmentType,
+					t2.visa';
 		if ($canViewClientInformation) {
 			$query .= ', phoneNumber, emailAddress ';
 		}
-		$query .= 'FROM Appointment
-					LEFT JOIN ServicedAppointment ON Appointment.appointmentId = ServicedAppointment.appointmentId
-					JOIN Client ON Appointment.clientId = Client.clientId
-					JOIN AppointmentTime ON Appointment.appointmentTimeId = AppointmentTime.appointmentTimeId
-				WHERE DATE(AppointmentTime.scheduledTime) = ?
-					AND AppointmentTime.siteId = ?
-					AND (cancelled IS NULL OR cancelled = FALSE)
-					AND (completed IS NULL OR completed != FALSE)
-					AND Appointment.archived = FALSE
-				ORDER BY AppointmentTime.scheduledTime, firstName, lastName';
+		$query .= 'FROM Appointment t1
+					LEFT JOIN ServicedAppointment ON t1.appointmentId = ServicedAppointment.appointmentId
+					JOIN Client ON t1.clientId = Client.clientId
+					JOIN AppointmentTime ON t1.appointmentTimeId = AppointmentTime.appointmentTimeId
+					JOIN AppointmentType ON AppointmentTime.appointmentTypeId = AppointmentType.appointmentTypeId
+					LEFT JOIN
+					(SELECT Answer.appointmentId, Answer.questionId, PossibleAnswer.text as visa
+					FROM Answer
+					JOIN PossibleAnswer ON Answer.possibleAnswerId = PossibleAnswer.possibleAnswerId
+					WHERE questionId = 3
```

Comment:
I'd prefer you use the lookupName that (I think?) exists in the question table. You'd probably need another sub-query here to find the corresponding id. There's probably precedent for this somewhere else in the codebase.

Reply:
Yeah, totally right. Probably frustrating for you to add the lookupName only for me to query by ID, sorry. I added that.